File: src/1404-Number-of-Steps-to-Reduce-a-Number-in-Binary-Representation-to-One/1404.cpp
class Solution {
public:
    int numSteps(string s) {
        int res = 0, add = 0, n = s.size();
        
        for (int i = n - 1; i; i--) {
            res++;
            if (s[i] - '0' + add == 1) {
                add = 1;
                res++;
            }
        }
        return res + add;
    }
};

File: src/1404-Number-of-Steps-to-Reduce-a-Number-in-Binary-Representation-to-One/1404.java
class Solution {
    public int numSteps(String s) {
        int res = 0, add = 0, n = s.length();
        
        for (int i = n - 1; i > 0; i--) {
            res++;
            if (s.charAt(i) - '0' + add == 1) {
                add = 1;
                res++;
            }
        }
        return res + add;
    }
}